Content.Shared/Whistle/WhistleSystem.cs

```diff
@@ -0,0 +1,61 @@
+using Content.Shared.Coordinates;
+using Content.Shared.Humanoid;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Stealth.Components;
+using Robust.Shared.Network;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Whistle;
+
+public abstract class SharedWhistleSystem : EntitySystem
+{
+    [Dependency] private readonly EntityLookupSystem _entityLookup = default!;
+    [Dependency] private readonly INetManager _netManager = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
+
+    private bool ExclamateTarget(EntityUid target, WhistleComponent component)
+    {
+        SpawnAttachedTo(component.Effect, target.ToCoordinates());
+
+        return true;
+    }
+
+    public void OnUseInHand(EntityUid uid, WhistleComponent component, UseInHandEvent args)
```

Comment:
Why don't you just make it private and subscribe to local event in Shared rather than doing it separately in Client and Server